### ISSUE TITLE
Web API command playPause does only pause

### ISSUE THREAD

**Issue by waltervolkers** [Mastermindzh/tidal-hifi#252]
My assumption is that the playPause feature exists within the Web API. When I try this, it is accepted but the result is only pause.

**Comment by Mastermindzh** [comment]
Are you on an album or playlist page?
Try going to the "home" page and see if it works there.

If it does -> known bug that I'm working on.
If it doesn't -> please tell me the steps to reproduce because mine works well :)

**Comment by waltervolkers** [comment]
Hoi Rick,

Ik neem aan dat je nederlands bent.
Wat ik gebruik is o.b.v. http get commando's de app remote te sturen.
Dat werkt erg goed. Zoals voor play http://localhost:47836/play
Alleen de playPause doet niet wat ik verwacht. toggle play pause

Walter

Op zo 23 jul 2023 om 22:41 schreef Rick van Lieshout <
***@***.***>:

**Comment by Mastermindzh** [comment]
Yes, I'm Dutch, see the Dutch reply below.

The issue might be that tidal-hifi is "stuck" on metadata parsing.
That bug has existed for some time and was "fixed" (for now...) in 5.4.0 which is yet to be released (will happen today).

Making sure you navigate back to the "homepage" skips the code that looks for album names in the table which causes the parser to get stuck (the underlying HTML changed, it can't find the albums).

I have attached a video (at the bottom) of it working in version 5.3.0 and also verified that this works on album/playlist pages in 5.4.0.

TL;DR: I can only replicate it in certain conditions. Most likely linked to an existing bug that will be fixed in 5.4.0 which will be released later today.

Dutch:

Yep, ik ben Nederlands.
Het zou kunnen dat tidal-hifi vast zit in het parsen van de metadata van het huidige nummer.
Die bug zit er al een tijdje in en is gefixed voor nu) in versie 5.4.0 die nog moet uitkomen (later vandaag).

Teruggaan naar de homepage skipped het codeblok dat albumnamen ophaalt uit de tabel die op de pagina staat, die tabel zorgt ervoor dat de metadata parser vast komt te zitten. (de onderliggende HTML is ooit aangepast door Tidal, hij vindt nu geen albumnamen meer)

Ik heb hieronder een video toegevoegd waar het werkt in 5.3.0 (op de homepage) en heb in 5.4.0 ook getest op album/playlist paginas.

TL;DR: Ik kan het gedrag alleen nabootsen onder specifieke condities. Dit is waarschijnlijk gelinked aan een bestaande bug die gefixed zal worden in versie 5.4.0 die later vandaag uitkomt.

Video:
https://github.com/Mastermindzh/tidal-hifi/assets/4190290/bf7838c0-d485-4e4f-a132-2f6f3726fade

**Comment by Mastermindzh** [comment]
I did not know GitHub couldn't handle the sound in the video, sorry.
You'll have to trust that the sound actually stops 😆

**Comment by waltervolkers** [comment]
Sound works !

I use flatpak

flatpak list
Name             Application ID                          Version    Branch
     Installation
tidal-hifi       com.mastermindzh.tidal-hifi             5.3.0      stable
     system

Walter

Op ma 24 jul 2023 om 11:41 schreef Rick van Lieshout <
***@***.***>:

**Comment by Mastermindzh** [comment]
I do not use flatpak so I haven't tested that. 
But if play and pause work separately then they should work together as well.

Mind trying the (non-flatpak) (unpacked) version of tidal-hifi 5.4.0?
You can find it here: https://files.mastermindzh.tech/f/98c11e85e7ff49c888ec/

Simply unzip, make the tidal-hifi executable executable (`chmod +x tidal-hifi`) and run it "./tidal-hifi"

**Comment by waltervolkers** [comment]
It is working now :-)
I will use the unpacked version of tidal-hifi 5.4.0 now



Op ma 24 jul 2023 om 12:14 schreef Rick van Lieshout <
***@***.***>:

**Comment by waltervolkers** [comment]
Issue has been resolved
